include/zephyr/dt-bindings/input/keymap.h

```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DT_BINDINGS_INPUT_KEYMAP_H_
+#define ZEPHYR_INCLUDE_DT_BINDINGS_INPUT_KEYMAP_H_
+
+#define MATRIX_KEY(row, col, code) \
+	((((row) & 0xff) << 24) | (((col) & 0xff) << 16) | ((code) & 0xffff))
```

Comment:
Yeah but that's a 16 bit mask, means you can have 16 rows, not 65 thousands. :-) The hard limit at this point are for the library 16x256 (https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/input/input_kbd_matrix.h#L138-L139) and when using the GPIO driver 16x32 (https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/input/input_gpio_kbd_matrix.c#L280).

I think it could be easily extended to 64x64, and I have to say it's quite tempting to implement that but then I'd love to make a board with the full 4096 buttons and prove it works.

![x](https://github.com/zephyrproject-rtos/zephyr/assets/891546/009e1289-b0b5-4a50-9027-089cc49faa19)

Yeah, nevermind.